controller/workitem.go

```diff
@@ -110,10 +111,42 @@ func (c *WorkitemController) List(ctx *app.ListWorkitemContext) error {
 		exp = criteria.And(exp, criteria.Equals(criteria.Field(workitem.SystemState), criteria.Literal(string(*ctx.FilterWorkitemstate))))
 		additionalQuery = append(additionalQuery, "filter[workitemstate]="+*ctx.FilterWorkitemstate)
 	}
+	if ctx.FilterCategory != nil {
+		var relationships []*category.WorkItemTypeCategoryRelationship
+		err := application.Transactional(c.db, func(tx application.Application) error {
+			// Load all workitemtypes related to the specific category
+			relationships, err = tx.Categories().LoadAllRelationshipsOfCategory(ctx, *ctx.FilterCategory)
+			if err != nil {
+				log.Error(ctx, map[string]interface{}{
+					"category_id": *ctx.FilterCategory,
+					"err":         err,
+				}, "failed to list work items in category")
+				return err
+			}
+			return nil
```

Comment:
@DhritiShikhar in a3fed8e you removed the if block but kept the logging of the error variable. That is not correct. We should only log errors when the error object is not nil. Please move the logging to the if block right below where you check for `err != nil`. Thank you.